### DISCUSSION TITLE
Builld failure on Opensuse - where to discuss?

### DISCUSSION THREAD

**Discussion by NelkonParker** [xournalpp/xournalpp#3478] [General] [answered]
I'm trying and failing to build Xournalpp from source on Opensuse leap.

Where is the most appropriate place to discuss this kind of problem? I can't find any development related problems in the Discussions section, so I guess this goes on elsewhere.

**Comment by LittleHuba** [comment]
Could you go into more detail what goes wrong?

**Reply by NelkonParker** [reply]
I've attached one file showing the contents of my CMakeCache.txt file (I think this is the most appropriate?) and also one with the output from cmake.

Let me know if you need any more info.

[CMakeCache.txt](https://github.com/xournalpp/xournalpp/files/7355070/CMakeCache.txt)

[cmake.txt](https://github.com/xournalpp/xournalpp/files/7355072/cmake.txt)

**Reply by Technius** [reply]
This looks like a bug of some sort. Please submit an issue.

**Reply by NelkonParker** [reply]
Does this not look more like a configuration issue than a bug? I 'd guess that the source must have been compiled on pretty much all flavours of Linux by now. 

I was assuming that I have some include path misconfigured for some reason, but it's been many years since I've done any software development, and I'm not too familiar with cmake, so I'm having difficulty seeing what to look at next.

I have the following gcc version, by the way:

/usr/bin/c++ --version
c++ (SUSE Linux) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Comment by LittleHuba** [comment, marked as answer]
It's the missing namespace syndrom again. We had this with `abs` quite a few times already. The issue is the compiler + linker that handle this slightly different on different OSs. @Technius is correct, please file a bug report. You may also add the `std::` namespace yourself and file a PR that fixes builds on OpenSuse.

**Reply by NelkonParker** [reply]
OK, thanks, I shall file a bug report. However, it's unlikely that I will be able to look into this any further for the next couple of weeks, so I won't be able to do it before the end of October.
